### ISSUE TITLE
New user workflow - trying to send message from android before any apps are define on server crashes app

### ISSUE THREAD

**Issue by parktheredcar** [gotify/android#178]
Trying to push a message out from the android app before any applications are defined on the gotify server causes the android app to hang until android ui offers to kill it.

Once an application has been defined on the server, the app dropdown on android's push message screen can no longer be set to a null selection, and it is no longer possible to trigger this issue through the app's ui. 

As a side effect, if the user accepts android's "kill hanging app" prompt, this also can put the gotify app to get in a state where the background listener is dead despite the main gotify app running in the foreground. This can lead to the app seeming unable to receive push notifications for a new user,

**Comment by jmattheis** [comment]
Good catch. The service currently tries to obtain the application image from the application. If there is a new application and the cache isn't refreshed, then there should be some kind of default instead of an error.

https://github.com/gotify/android/blob/0a7a04885ff7a6463ccbf48f2e7742a11cf977a9/app/src/main/java/com/github/gotify/picasso/PicassoHandler.java#L59-L73
